### PR TITLE
Move more parts of make check to mage

### PR DESCRIPTION
**Pull request by jsoriano**
Use mage implementation for make check in directories
where make is still being used. Fix issues detected in
dashboards and in dev-tools directory, that was not
being checked. `DashboardsFormat` renamed to
`CheckDashboardsFormat` to be more explicit and
coherent with other check sub-targets.

Continues dashboard checking started in #13168.

License header targets are not migrated here, migration
in progesss in #13215.

### CODE REVIEW

**Comment by jsoriano** [comment]
> I just realized now all the dashboard screenshots are not up to date grimacing Just the names of the visualizations are different so hopefully it's ok.

I also hope this is ok :grimacing: I wouldn't change all the screenshots now.

Dashboards using custom titles for visualizations wouldn't be affected, but not all of them are using this feature.

**Comment by andrewkroh** [comment]
I left a similar comment in #13215, I think it would be nice to add this to dev-tools/mage/target/common and then update each project to 

```
import (
...

	// mage:import
	_ "github.com/elastic/beats/dev-tools/mage/target/common"
)
```

which will help reduce duplication.

**Comment by jsoriano** [comment]
@andrewkroh I had tried, but I found some conflicting targets, and that `Check` depended on `Update` in some magefiles, so I postponed it. But now checking again I have discovered `RegisterCheckDeps()` to inject the `Update`s. So with this, and removing the conflicting targets, all good :slightly_smiling_face:

All magefiles use the common targets now.

Once we have this merged I will update #13215, that has more things.

File: filebeat/magefile.go [hunk 1]
```diff
@@ -28,9 +28,14 @@ import (
 
 	devtools "github.com/elastic/beats/dev-tools/mage"
 	filebeat "github.com/elastic/beats/filebeat/scripts/mage"
+
+	// mage:import
+	common "github.com/elastic/beats/dev-tools/mage/target/common"
```
[the remark this answers is not in the document]
Comment (jsoriano) [reply filebeat/magefile.go]: Oh right :sweat_smile:

**Comment by jsoriano** [comment]
@andrewkroh there were a couple of legit issues in CI, I think I have solved them. One was caused by the different python envs mage and make use under the same `PYTHON_ENV`. This may need another quick review.

File: libbeat/scripts/Makefile [hunk 2]
```diff
@@ -241,7 +241,7 @@ load-tests: ## @testing Runs load tests
 # Sets up the virtual python environment
 .PHONY: python-env
 python-env: ${ES_BEATS}/libbeat/tests/system/requirements.txt
-	@test -d ${PYTHON_ENV} || virtualenv $(if ${PYTHON_EXE},-p ${PYTHON_EXE}) ${VIRTUALENV_PARAMS} ${PYTHON_ENV}
+	@test -d ${PYTHON_ENV}/bin/activate || virtualenv $(if ${PYTHON_EXE},-p ${PYTHON_EXE}) ${VIRTUALENV_PARAMS} ${PYTHON_ENV}
```
Comment (andrewkroh) [inline libbeat/scripts/Makefile]: -d is specifically checking for a directory so probably -f (file) or just -e (exists) need to be used instead.

Comment (jsoriano) [reply libbeat/scripts/Makefile]: Good catch :+1:

**Comment by jalvz** [comment]
Hey, this change broke apm-server.
I understand that it can be hard to _remember_ what might break apm-server, but https://beats-ci.elastic.co/job/elastic+apm-server+master+beats-update/ runs periodically on master and should provide feedback.
Now, it looks like that job is permanently broken... did anyone raise the alarm, or it is not really being used?

Would it be better to run that job on every pr touching libbeat code instead?

**Comment by jsoriano** [comment]
Hey @jalvz,

I am very sorry. Actually I didn't know (or remember) that there was a build of apm-server running with the last version of libbeat.

I can take a look to what is failing after this PR if nobody is working on it, or I can help if someone is already on it.

> Would it be better to run that job on every pr touching libbeat code instead?

This would be something to discuss, but it would be definitely helpful to have some way of knowing when libbeat breaks apm-server builds.

**Comment by jalvz** [comment]
No worries, that is understandable - it is why i suggest to run it on libbeat prs (also for future beats contributors, etc)

> it would be definitely helpful to have some way of knowing when libbeat breaks apm-server builds.

The conservative answer is that breaking changes on a libbeat Makefile or exported functions might break us. It practical terms, most of the times it happens with make / mage changes.

The only automated way we have to check is by running on beats CI the same command we run locally to update beats. Adding that stage would bump build times ~15 minutes, that is why you have it running only on master - but then it can get forgotten about.

Maybe a good middle ground is to trigger that stage only upon changes on `devtools/mage*` and `libbeat/scripts/Makefile`

What do you think?

**Comment by jsoriano** [comment]
> The only automated way we have to check is by running on beats CI the same command we run locally to update beats. Adding that stage would bump build times ~15 minutes, that is why you have it running only on master - but then it can get forgotten about.

I guess this can be run in parallel, so it doesn't affect the total time of builds.

> Maybe a good middle ground is to trigger that stage only upon changes on `devtools/mage*` and `libbeat/scripts/Makefile`
> 
> What do you think?

That sounds reasonable, this way we'd force ourselves to make changes in a more backwards compatible way.

**Pull request merged**